### ISSUE TITLE
Fix S3655 FP: Nullable of a ValueTuple and ??=

### ISSUE THREAD

**Issue by antonioaversa** [SonarSource/sonar-dotnet#7089]
Fix the false positive on S3655 found on [peach](https://peach.sonarsource.com/code?id=ravendb&selected=ravendb:src/Raven.Server/Documents/ETL/Providers/Raven/RavenEtlDocumentTransformer.cs&line=664):
```cs
foreach (LazyStringValue timeSeriesName in timeSeriesNames)
{
    (DateTime from, DateTime to)? toLoad = null;
    if(loadBehaviorFunction != null && ShouldFilterByScriptAndGetParams(item.DocumentId, timeSeriesName, loadBehaviorFunction, out toLoad))
        continue;
    toLoad ??= (DateTime.MinValue, DateTime.MaxValue);
    
    var reader = Database.DocumentsStorage.TimeSeriesStorage.GetReader(Context, item.DocumentId, timeSeriesName, toLoad.Value.from, toLoad.Value.to); // Noncompliant {{'toLoad' is null on at least one execution path.}}, FP
    var deletedRanges = Database.DocumentsStorage.TimeSeriesStorage.GetDeletedRangesForDoc(Context, item.DocumentId);
    ret[timeSeriesName] = new TimeSeriesModifications(reader.AllValues(), deletedRanges);
}
```

Documented in UT via [this PR](https://github.com/SonarSource/sonar-dotnet/pull/7029).

The specific scenario discussed [here](https://github.com/SonarSource/sonar-dotnet/pull/7029#discussion_r1163032114).

**Comment by martin-strecker-sonarsource** [comment]
Tracked by https://sonarsource.atlassian.net/browse/NET-1199 now.
